spring-cloud-gcp-spanner/src/main/java/org/springframework/data/spanner/core/mapping/BasicSpannerPersistentEntity.java

```diff
@@ -0,0 +1,90 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.data.spanner.core.mapping;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.springframework.data.mapping.model.BasicPersistentEntity;
+import org.springframework.data.util.TypeInformation;
+import org.springframework.util.StringUtils;
+
+/**
+ * Represents a Spanner table and its columns' mapping to fields within an entity type.
+ *
+ * @author Ray Tsang
+ * @author Chengyuan Zhao
+ */
+public class BasicSpannerPersistentEntity<T>
+		extends BasicPersistentEntity<T, SpannerPersistentProperty>
+		implements SpannerPersistentEntity<T> {
+
+	private final String tableName;
+
+	private final Set<String> columnNames = new HashSet<>();
+
+	private final Map<String, String> columnNameToPropertyName = new HashMap<>();
+
+	/**
+	 * Constructor
+	 * @param information type information about the underlying entity type.
+	 */
+	public BasicSpannerPersistentEntity(TypeInformation<T> information) {
+		super(information);
+
+		Class<?> rawType = information.getType();
+		String fallback = StringUtils.uncapitalize(rawType.getSimpleName());
+
+		Table table = this.findAnnotation(Table.class);
+		if (table != null) {
+			this.tableName = StringUtils.hasText(table.name()) ? table.name() : fallback;
```

Comment:
the null check and the hasText can be collapsed -- the if is completely unnecessary